ui-v2/src/components/flow-runs/flow-run-state-dialog/use-flow-run-state-dialog.tsx

```diff
@@ -0,0 +1,72 @@
+import type { FlowRun } from "@/api/flow-runs";
+import { useSetFlowRunState } from "@/api/flow-runs";
+import { RUN_STATES } from "@/api/flow-runs/constants";
+import type { components } from "@/api/prefect";
+import type { RunStateFormValues } from "@/components/ui/run-state-change-dialog";
+import { StateBadge } from "@/components/ui/state-badge";
+import { useCallback, useState } from "react";
+import { toast } from "sonner";
+
+export const useFlowRunStateDialog = () => {
+	const [open, setOpen] = useState(false);
+	const [flowRun, setFlowRun] = useState<FlowRun | null>(null);
+	const { mutateAsync: setFlowRunState } = useSetFlowRunState();
```

Comment:
```suggestion
	const { setFlowRunState } = useSetFlowRunState();
```
suggestion: use the `mutate` version `setFlowRunState`

Reply:
done! [927ee96](https://github.com/PrefectHQ/prefect/pull/17704/commits/927ee9610f0e103c75e2de31b71febe4a5730d22)